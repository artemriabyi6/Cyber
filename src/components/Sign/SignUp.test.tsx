import React from 'react';
import { renderToString } from 'react-dom/server';
import SignUpForm from './SignUp';

// Mock для next-auth
jest.mock('next-auth/react', () => ({
  signIn: jest.fn(() => Promise.resolve({ error: null, ok: true })),
  useSession: jest.fn(() => ({ data: null, status: 'unauthenticated' })),
}));

// Mock для next/navigation
jest.mock('next/navigation', () => ({
  useRouter: () => ({
    push: jest.fn(),
  }),
}));



// Mock для fetch
global.fetch = jest.fn();

describe('SignUpForm Component', () => {
  beforeEach(() => {
    jest.clearAllMocks();
    (global.fetch as jest.Mock).mockClear();
  });

  test('рендериться без помилок', () => {
    const htmlString = renderToString(<SignUpForm />);
    expect(htmlString).toBeTruthy();
  });

  test('містить заголовок форми реєстрації', () => {
    const htmlString = renderToString(<SignUpForm />);
    expect(htmlString).toContain('Створіть акаунт');
    expect(htmlString).toContain('Заповніть форму для реєстрації');
  });

  test('містить всі необхідні поля форми', () => {
    const htmlString = renderToString(<SignUpForm />);
    
    expect(htmlString).toContain('Ім&#x27;я');
    expect(htmlString).toContain('Прізвище');
    expect(htmlString).toContain('Email');
    expect(htmlString).toContain('Пароль');
    expect(htmlString).toContain('Підтвердіть пароль');
  });

  test('поля імені та прізвища знаходяться в grid контейнері', () => {
    const htmlString = renderToString(<SignUpForm />);
    expect(htmlString).toContain('grid grid-cols-2 gap-4');
  });

  test('містить кнопку реєстрації', () => {
    const htmlString = renderToString(<SignUpForm />);
    expect(htmlString).toContain('Зареєструватись');
  });

  test('містить посилання на вхід', () => {
    const htmlString = renderToString(<SignUpForm />);
    expect(htmlString).toContain('Вже маєте акаунт?');
    expect(htmlString).toContain('Увійти');
    expect(htmlString).toContain('href="/signin"');
  });

  test('поле пароля має мінімальну довжину', () => {
    const htmlString = renderToString(<SignUpForm />);
    expect(htmlString).toContain('minLength="6"');
  });

  test('всі поля форми мають required атрибут', () => {
    const htmlString = renderToString(<SignUpForm />);
    // Перевіряємо що є кілька required атрибутів
    const requiredCount = (htmlString.match(/required/g) || []).length;
    expect(requiredCount).toBeGreaterThan(3);
  });

  test('відповідає snapshot', () => {
    const htmlString = renderToString(<SignUpForm />);
    expect(htmlString).toMatchSnapshot();
  });
});

describe('SignUpForm Structure', () => {
  test('має правильну структуру класів Tailwind', () => {
    const htmlString = renderToString(<SignUpForm />);
    
    expect(htmlString).toContain('bg-gradient-to-br');
    expect(htmlString).toContain('from-blue-50');
    expect(htmlString).toContain('to-indigo-100');
    expect(htmlString).toContain('min-h-screen');
    expect(htmlString).toContain('max-w-md');
    expect(htmlString).toContain('bg-white');
    expect(htmlString).toContain('rounded-2xl');
  });

  test('кнопка має правильні стилі та анімації', () => {
    const htmlString = renderToString(<SignUpForm />);
    
    expect(htmlString).toContain('bg-gradient-to-r');
    expect(htmlString).toContain('from-blue-600');
    expect(htmlString).toContain('to-indigo-600');
    expect(htmlString).toContain('transform');
    expect(htmlString).toContain('hover:-translate-y-0.5');
    expect(htmlString).toContain('transition-all');
  });

  test('поля вводу мають правильні стилі фокусу', () => {
    const htmlString = renderToString(<SignUpForm />);
    
    expect(htmlString).toContain('focus:ring-2');
    expect(htmlString).toContain('focus:ring-blue-500');
    expect(htmlString).toContain('focus:border-transparent');
  });
});

describe('SignUpForm Accessibility', () => {
  test('всі поля мають правильні labels', () => {
    const htmlString = renderToString(<SignUpForm />);
    
    expect(htmlString).toContain('for="firstName"');
    expect(htmlString).toContain('for="lastName"');
    expect(htmlString).toContain('for="email"');
    expect(htmlString).toContain('for="password"');
    expect(htmlString).toContain('for="confirmPassword"');
  });

  test('поля мають правильні placeholder', () => {
    const htmlString = renderToString(<SignUpForm />);
    
    expect(htmlString).toContain('placeholder="Введіть ім&#x27;я"');
    expect(htmlString).toContain('placeholder="Введіть прізвище"');
    expect(htmlString).toContain('placeholder="your@email.com"');
    expect(htmlString).toContain('placeholder="Мінімум 6 символів"');
    expect(htmlString).toContain('placeholder="Повторіть пароль"');
  });
});