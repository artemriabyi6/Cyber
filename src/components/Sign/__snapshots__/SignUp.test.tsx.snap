// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`SignUpForm Component відповідає snapshot 1`] = `"<div class="min-h-screen bg-gradient-to-br from-blue-50 to-indigo-100 flex items-center justify-center px-4 py-8"><div class="max-w-md w-full space-y-8"><div class="bg-white rounded-2xl shadow-xl p-8"><div class="text-center mb-8"><h2 class="text-3xl font-bold text-gray-900">Створіть акаунт</h2><p class="mt-2 text-gray-600">Заповніть форму для реєстрації</p></div><form class="space-y-6"><div class="grid grid-cols-2 gap-4"><div><label for="firstName" class="block text-sm font-medium text-gray-700 mb-1">Ім&#x27;я</label><input id="firstName" type="text" required="" class="w-full px-4 py-3 border border-gray-300 rounded-lg focus:ring-2 focus:ring-blue-500 focus:border-transparent transition-all" placeholder="Введіть ім&#x27;я" name="firstName" value=""/></div><div><label for="lastName" class="block text-sm font-medium text-gray-700 mb-1">Прізвище</label><input id="lastName" type="text" required="" class="w-full px-4 py-3 border border-gray-300 rounded-lg focus:ring-2 focus:ring-blue-500 focus:border-transparent transition-all" placeholder="Введіть прізвище" name="lastName" value=""/></div></div><div><label for="email" class="block text-sm font-medium text-gray-700 mb-1">Email</label><input id="email" type="email" required="" class="w-full px-4 py-3 border border-gray-300 rounded-lg focus:ring-2 focus:ring-blue-500 focus:border-transparent transition-all" placeholder="your@email.com" name="email" value=""/></div><div><label for="password" class="block text-sm font-medium text-gray-700 mb-1">Пароль</label><input id="password" type="password" required="" minLength="6" class="w-full px-4 py-3 border border-gray-300 rounded-lg focus:ring-2 focus:ring-blue-500 focus:border-transparent transition-all" placeholder="Мінімум 6 символів" name="password" value=""/></div><div><label for="confirmPassword" class="block text-sm font-medium text-gray-700 mb-1">Підтвердіть пароль</label><input id="confirmPassword" type="password" required="" class="w-full px-4 py-3 border border-gray-300 rounded-lg focus:ring-2 focus:ring-blue-500 focus:border-transparent transition-all" placeholder="Повторіть пароль" name="confirmPassword" value=""/></div><button type="submit" class="w-full bg-gradient-to-r from-blue-600 to-indigo-600 text-white py-3 px-4 rounded-lg font-semibold hover:from-blue-700 hover:to-indigo-700 transform hover:-translate-y-0.5 transition-all duration-200 shadow-lg hover:shadow-xl disabled:opacity-50 disabled:cursor-not-allowed">Зареєструватись</button><div class="text-center"><p class="text-gray-600">Вже маєте акаунт?<!-- --> <a class="text-blue-600 hover:text-blue-500 font-medium" href="/signin">Увійти</a></p></div></form></div></div></div>"`;
